omas/omas_physics.py

```diff
@@ -29,6 +29,29 @@ def add_to__ALL__(f):
 class constants(object):
     e = 1.6021766208e-19
 
+@add_to__ODS__
+def equilibrium_stored_energy(ods, update=True):
+    '''
+    Calculate MHD stored energy from equilibrium pressure and volume
+
+    :param ods: input ods
+
+    :param update: operate in place
+
+    :return: updated ods
+    '''
+    ods_n = ods
+    if not update:
+        from omas import ODS
+        ods_n = ODS().copy_attrs_from(ods)
+
+    for time_index in ods['equilibrium']['time_slice']:
+        pressure_equil = ods['equilibrium']['time_slice'][time_index]['profiles_1d']['pressure']
+        volume_equil = ods['equilibrium']['time_slice'][time_index]['profiles_1d']['volume']
+        dvol = numpy.gradient(volume_equil)
```

Comment:
Why calculate `dvol`?  Why not just give `volume_equil` to `numpy.trapz` as the keyword `x`?  (see https://numpy.org/doc/1.18/reference/generated/numpy.trapz.html#r7aa6c77779c0-2)